openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainData.java

```diff
@@ -118,6 +159,18 @@ public void addLabelValue(float value) {
         this.swigLabelsChunkedArray.add(value);
```

Comment:
This is exposed publicly and it is quite dangerous to have memory being written outside of the array allocated space.

if you run this code
```
final SWIGTrainData data = new SWIGTrainData(2,2, true);
data.addFeatureValue(1d);
data.addFeatureValue(2d);
data.addFeatureValue(3d); // after this call there is a probability that your VM will crash with a something like `munmap_chunk(): invalid pointer`
```
Same may happen to labels

Reply:
is it ok if we set everything to package private? or should we add checks to see if the size of the array does not exceed allocated memory?

Reply:
This won't do any harm as it dynamically keeps allocating memory with new chunks if need be.

However, if one calls a different number of times `addFeatureValue` than `addLabelValue` for instance, our dataset wouldn't make sense.

Reply:
action: add assert to check `#feature_adds / #features/instance = #label_adds = #constraint_group_adds` before using the data